tests/test_conv.rs

```diff
@@ -0,0 +1,36 @@
+use neural_zkp::conv::*;
+
+use ndarray::{Array3, Array4};
+use ndarray_rand::{rand::SeedableRng, rand_distr::Uniform, RandomExt};
+use rand_isaac::isaac64::Isaac64Rng;
+
+#[test]
+fn conv_test() {
+    // man of culture fixed randomness seed
+    let seed = 694201337;
+    let mut rng = Isaac64Rng::seed_from_u64(seed);
+
+    let input = Array3::random_using((120, 80, 3), Uniform::<f32>::new(-5., 5.), &mut rng);
+    let kernel = Array4::random_using((32, 5, 5, 3), Uniform::<f32>::new(-10., 10.), &mut rng);
+
+    // x is the result of  conv(input, kernel)
+    let Conv2D::<f32> {
+        output: x,
+        n_params,
+        n_multiplications,
+        name,
+    } = convolution(input, kernel);
+
+    assert_eq!(x.dim(), (116, 76, 32));
```

Comment:
Is this the only correctness check we do?

Reply:
yes, since the input is random, there is no way to objectively verify the output besides running it in python, exporting, importing and checking against rust output.